userspace/libsinsp/dumper.cpp

```diff
@@ -47,13 +57,20 @@ void sinsp_dumper::open(const string& filename, bool compress, bool threads_from
 		m_inspector->m_thread_manager->to_scap();
 	}
 
-	if(compress)
+	if(m_target_memory_buffer)
```

Comment:
It looks like you're using m_target_memory_buffer non null == memory dump. Do you want to have an explicit type perhaps reusing ppm_dumper_type?